staging/src/k8s.io/apiserver/pkg/storage/tests/cacher_test.go

```diff
@@ -268,42 +276,6 @@ func TestList(t *testing.T) {
 	}
 }
 
-// TestTooLargeResourceVersionList ensures that a list request for a resource version higher than available
```

Comment:
why are you removing this test?

Reply:
also, shouldn't we have a test that ensures that the list from the watch-cache returns keys sorted, since you are basically fixing that?

Reply:
> why are you removing this test?

That's the broader point of the PR - to re-use the tests for this exact behavior we have for etcd layer. It's being tested now in the `TestList()` delegated call to the storage testing library.

> also, shouldn't we have a test that ensures that the list from the watch-cache returns keys sorted, since you are basically fixing that?

+1 This is currently implicit in the way the etcd test is written, but we might want to either explicitly call it out in the test names or write a supplemental test that sorts the results itself and validates that it got a sorted list.

Reply:
Pushed the change in additional separate commit - PTAL